packages/contracts/src/dollar/core/TWAPOracleDollar3pool.sol

```diff
@@ -1,31 +1,28 @@
 // SPDX-License-Identifier: Apache-2.0
-pragma solidity ^0.8.3;
+pragma solidity 0.8.16;
 
 import "../interfaces/IMetaPool.sol";
+import "../interfaces/ITWAPOracleDollar3pool.sol";
 
-contract TWAPOracleDollar3pool {
+contract TWAPOracleDollar3pool is ITWAPOracleDollar3pool {
     address public immutable pool;
     address public immutable token0;
     address public immutable token1;
-    uint256 public price0Average;
-    uint256 public price1Average;
+    uint256 public uADPrice;
```

Comment:
Old changes comment again. Perhaps this should be UbiquityDollarPrice given that its dealing with other protocol tokens as well in this code.

Reply:
done